src/tests/ftest/pytorch/dataset.py

```diff
@@ -0,0 +1,249 @@
+"""
+  (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+  (C) Copyright 2025 Google LLC
+  (C) Copyright 2025 Enakta Labs Ltd
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import hashlib
+
+from apricot import TestWithServers
+from dfuse_utils import get_dfuse, start_dfuse
+from io_utilities import DirectoryTreeCommand
+from pydaos.torch import Dataset, IterableDataset
+from run_utils import run_remote
+from torch.utils.data import DataLoader
+
+
+class PytorchDatasetsTest(TestWithServers):
+    """Test Pytorch Map Style Dataset.
+
+    :avocado: recursive
+    """
+
+    def test_map_style_dataset(self):
+        """Test Map Style Dataset directly without DataLoader
+
+        Test Description: Ensure that the dataset can read all the samples that were seeded.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=dfuse,pytorch
+        :avocado: tags=PytorchDatasetsTest,test_map_style_dataset
+        """
+        pool = self.get_pool()
+        container = self.get_container(pool)
+        dfuse = get_dfuse(self, self.hostlist_clients)
+        start_dfuse(self, dfuse, pool, container)
+
+        root_dir = dfuse.mount_dir.value
+
+        height = self.params.get("tree_height", "/run/map_style_dataset/*")
+        subdirs = self.params.get("subdirs", "/run/map_style_dataset/*")
+        files_per_node = self.params.get("files_per_node", "/run/map_style_dataset/*")
+        file_min_size = self.params.get("file_min_size", "/run/map_style_dataset/*", 4096)
+        file_max_size = self.params.get("file_max_size", "/run/map_style_dataset/*", 128 * 1024)
+
+        self._create_test_files(root_dir, height, subdirs, files_per_node,
+                                file_min_size, file_max_size)
+
+        expected = self._get_test_files_hashmap(root_dir, self.hostlist_clients)
+
+        dataset = Dataset(pool.identifier, container.identifier)
+
+        actual = {}
+        for _, content in enumerate(dataset):
+            h = hashlib.md5(content).hexdigest()
```

Comment:
Python bandit is complaining with
> Use of weak MD5 hash for security. Consider usedforsecurity=False

This is test code so not a real concern, but the easiest way to silence is to resolve
```suggestion
            h = hashlib.md5(content, usedforsecurity=False).hexdigest()
```